### PR TITLE
313 implement nuc sequence aligned

### LINKED ISSUE #312: Implement: `/aa-sequence-aligned/{gene}`
Similar to LAPIS v1

### LINKED ISSUE #313: Implement `/nuc-sequence-aligned`
Similar to LAPIS v1

Also have `/fasta-aligned` map to it

### LINKED ISSUE #314: Implement `/nuc-sequence`
Similar to LAPIS v1

`/fasta` should also map to this endpoint

### PR DESCRIPTION
**Pull request by JonasKellerer**
Resolves #313 resolves #312

### CODE REVIEW

**Comment by JonasKellerer** [comment]
This PR waits on SILO to implement the "Fasta" action.

File: lapis2/src/main/kotlin/org/genspectrum/lapis/OpenApiDocs.kt [hunk 1]
```diff
@@ -162,6 +171,12 @@ fun buildOpenApiSchema(sequenceFilterFields: SequenceFilterFields, databaseConfi
                             .properties(aminoAcidInsertionSchema()),
                     ),
                 )
+                .addSchemas(
+                    ALIGNED_NUCLEOTIDE_SEQUENCE_RESPONSE_SCHEMA,
+                    lapisResponseSchema(
+                        Schema<String>().description("Aligned nucleotide sequence data."),
+                    ),
+                )
```
Comment (fengelniederhammer) [inline lapis2/src/main/kotlin/org/genspectrum/lapis/OpenApiDocs.kt:179]: Is this even used? Anyway using the general LAPIS schema seems wrong here.

File: lapis2/src/main/kotlin/org/genspectrum/lapis/controller/SingleSegmentedSequenceController.kt [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import org.genspectrum.lapis.request.SequenceFiltersRequest
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+
+private const val ALIGNED_SINGLE_SEGMENTED_NUCLEOTIDE_SEQUENCE_ENDPOINT_DESCRIPTION =
+    "Returns a string of fasta formated aligned nucleotide sequences. Only sequences matching the " +
```
Comment (fengelniederhammer) [inline lapis2/src/main/kotlin/org/genspectrum/lapis/controller/SingleSegmentedSequenceController.kt:26]:
```suggestion
    "Returns a string of fasta formatted, aligned nucleotide sequences. Only sequences matching the " +
```

File: lapis2/src/main/kotlin/org/genspectrum/lapis/controller/MultiSegmentedSequenceController.kt [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+
+private const val ALIGNED_MULTI_SEGMENTED_NUCLEOTIDE_SEQUENCE_ENDPOINT_DESCRIPTION =
+    "Returns a string of fasta formated aligned nucleotide sequences of the requested segment. " +
```
Comment (fengelniederhammer) [inline lapis2/src/main/kotlin/org/genspectrum/lapis/controller/MultiSegmentedSequenceController.kt:26]:
```suggestion
    "Returns a string of fasta formatted, aligned nucleotide sequences of the requested segment. " +
```

File: siloLapisTests/test/common.ts [hunk 4]
```diff
@@ -23,3 +23,27 @@ export const lapisClient = new LapisControllerApi(new Configuration({ basePath }
...
+  post: async responseContext => {
+    if (responseContext.response.status >= 300) {
+      const response = responseContext.response.clone();
+      console.log('Request URL:', responseContext.url);
+      console.log('Response status code:', response.status);
+      console.log('Response body:', await response.json());
+    }
+    return Promise.resolve(responseContext.response);
+  },
+});
```
Comment (fengelniederhammer) [inline siloLapisTests/test/common.ts:49]: We could also extract the middleware, since it's the same as for the `lapisClient`, isn't it?

**Review by fengelniederhammer — CHANGES_REQUESTED** [review]
Some details, but overall it looks good :+1:

**Review by fengelniederhammer — APPROVED** [review]
Only some typos, etc. Otherwise it looks good :)

**Pull request merged**